nipoppy/zenodo.py

```diff
@@ -41,30 +52,38 @@ def download_record_files(self, zenodo_id: str, output_dir: Path):
         InvalidChecksumError
             Checksum mismatch between the downloaded file and the expected checksum.
         """
+        zenodo_id = zenodo_id.removeprefix("zenodo.")
         output_dir.mkdir(parents=True, exist_ok=True)
+
         response = httpx.get(
             f"{self.api_endpoint}/records/{zenodo_id}/files",
             headers=self.headers,
         )
         if response.status_code != 200:
-            raise Exception(response.json())
+            raise ZenodoAPIError(f"Failed to get files for record: {zenodo_id}")
 
         # Exclude "md5:" prefix
         files = {
-            entry["key"]: entry["checksum"][4:] for entry in response.json()["entries"]
+            entry["key"]: entry["checksum"].removeprefix("md5:")
+            for entry in response.json()["entries"]
         }
         for filename, checksum in files.items():
             response = httpx.get(
                 f"{self.api_endpoint}/records/{zenodo_id}/files/{filename}/content",  # noqa E501
                 headers=self.headers,
             )
             if response.status_code != 200:
-                raise Exception(response.json())
+                raise ZenodoAPIError(
+                    f"Failed to download file for zenodo.{zenodo_id}: {filename}"
+                )
 
             # Checksum verification before writing to disk
             content_md5 = hashlib.md5(response.content).hexdigest()
             if content_md5 != checksum:
-                raise InvalidChecksumError(f"Checksum mismatch: {filename}")
+                raise InvalidChecksumError(
+                    "Checksum mismatch: "
+                    f"{filename} has invalid checksum ({content_md5})"
+                )
 
             with output_dir.joinpath(filename).open("wb") as f:
                 f.write(response.content)
```

Comment:
Not 100% this will work but we could try this here:
```suggestion
            output_dir.joinpath(filename).write_bytes(response.content)
```